pkg/image/manifest_test.go

```diff
@@ -164,21 +164,28 @@ func TestGetE2EImageTagPairs(t *testing.T) {
 	}
 	expectedDefaultRegistry := defaultRegistry.v1_17()
 	if len(imageTagPairs) != len(expectedDefaultRegistry) {
-		t.Fatalf("Unexpected number of image tag pairs returned, expected %v, got %v", len(expectedDefaultRegistry), len(imageTagPairs))
+		t.Fatalf("unexpected number of image tag pairs returned, expected %v, got %v", len(expectedDefaultRegistry), len(imageTagPairs))
 	}
 
-	// Check one of the returned image pairs to ensure correct format
-	imageTagPair := imageTagPairs[0]
-	if strings.HasPrefix(imageTagPair.Src, customRegistry) {
-		t.Errorf("Src image should not have custom registry prefix: %q", imageTagPair.Src)
+	// As a sample, check one of the images for E2E and assert their mapping
+	var e2eImageTagPair TagPair
+	for _, imageTagPair := range imageTagPairs {
+		if strings.Contains(imageTagPair.Src, "e2e") {
+			e2eImageTagPair = imageTagPair
+			break
+		}
+	}
+
+	if strings.HasPrefix(e2eImageTagPair.Src, customRegistry) {
```

Comment:
Also, if that loop never break (above) does that create any initializiation hazard for var `e2eImageTagPair`? It does not seem like it (since e2eimagetagpair is a struct) but check please.

Reply:
ah yeah that's a good point. I think `e2e` will always exist, but I also think it's valuable to have check to assert it.